build/rpmfc.c

```diff
@@ -1234,27 +1243,26 @@ rpmRC rpmfcClassify(rpmfc fc, ARGV_t argv, rpm_mode_t * fmode)
 	case S_IFLNK:
 	case S_IFREG:
 	default:
-	    /* XXX all files with extension ".pm" are perl modules for now. */
-	    if (rpmFileHasSuffix(s, ".pm"))
-		ftype = "Perl5 module source text";
-
-	    /* XXX all files with extension ".la" are libtool for now. */
-	    else if (rpmFileHasSuffix(s, ".la"))
-		ftype = "libtool library file";
-
-	    /* XXX all files with extension ".pc" are pkgconfig for now. */
-	    else if (rpmFileHasSuffix(s, ".pc"))
-		ftype = "pkgconfig file";
+	    /* Skip libmagic for some well known file extensions. To avoid the
+	       costly calculcation but also to have a stable output for some */
+	    while (skipped_extensions[extension_index]) {
+		if (rpmFileHasSuffix(s, skipped_extensions[extension_index++])) {
+	            ftype = skipped_extensions[extension_index++];
+		    fmime = skipped_extensions[extension_index++];
+		} else {
+		    extension_index += 2;
```

Comment:
Please use a struct for this data, it's soooo much more obvious and readable.

Reply:
Fine

Reply:
I updated this now to use a struct, but somehow github struggles to update the PR. I hope this resolves soonish, otherwise I'll create a new PR